### PR TITLE
Reorganize dxCompiler code into subprojects

### PR DESCRIPTION
**Pull request by jdidion**
* Splits codebase into 5 subprojects:
  - core: shared code between front-end and back-end
  - compiler: the front-end
  - executorCommon: shared code between executors
  - executorWdl: the WDL executor
  - executorCwl: the CWL executor
* A few classes were moved into different packages to clear up invalid dependencies (e.g. code in `core` depending on a class in `compiler`)
* Changes (almost) all references from dxWDL to dxCompiler
* Updates build and test scripts to handle the new layout
  - Changes the way the applet_resources folder is used:
    - Common binaries are stored in sub-folders by version
    - Each executor is built in its own sub-folder
* Updates to latest wdlTools (0.11.8)
* Note:
  - many test resources are duplicated because I can't figure out how to get sub-projects to share test resources
  - currently, a single version is used for all subprojects, located in core/src/main/resources/application.conf. All other subprojects have their own application.confs, but I'm not sure whether it's necessary to maintain the version information there; I have left it as-is for now.

### CODE REVIEW

File: compiler/src/main/resources/dxCompiler_runtime.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+dxCompiler {
+  regionToProject = [
+    {
+      region = "aws:us-east-1"
+      path = "dxCompiler_playground:/builds/jdidion/2.0.0-rc6"
```
Comment (commandlinegirl) [inline compiler/src/main/resources/dxCompiler_runtime.conf:5]: Was this added accidentally?

**Review by commandlinegirl — COMMENTED** [review]
Partially reviewed, just one question so far about `dxCompiler_runtime.conf`.

File: compiler/src/main/resources/dxCompiler_runtime.conf (hunk 1 above)
Comment (jdidion) [reply compiler/src/main/resources/dxCompiler_runtime.conf:5]: yes thanks for catching. I've updated .gitignore.

**Review by commandlinegirl — APPROVED** [review]

**Comment by commandlinegirl** [comment]
@jdidion Please include the JIRA ticket ID in the commit title. Thanks!

**Comment by commandlinegirl** [comment]
One more question: will anything need to change in the release process?

File: build.sbt [hunk 2]
```diff
@@ -1,43 +1,190 @@
...
+    ),
+    publishMavenStyle := true,
+    // Tests
+    // If an exception is thrown during tests, show the full
+    // stack trace, by adding the "-oF" option to the list.
+    Test / testOptions += Tests.Argument("-oF"),
+    Test / parallelExecution := false
+    // Coverage
+    //
+    // sbt clean coverage test
```
[the remark this answers is not in the document]
Comment (jdidion) [reply build.sbt:176]: The goal is to re-enable it at some poInt

**Comment by jdidion** [comment]
Not really any change to the release - there will be separate dxCompiler.jar and dxExecutor*.jar files, but it is only necessary to release the dxCompiler.jar on github.

**Pull request merged**